devel/retdec/files/patch-openssl-v40.diff

```diff
@@ -0,0 +1,49 @@
+commit a0da280a962abde45d248bd0d7d95580c671f160
+Author: Kirill A. Korinsky <kirill@korins.ky>
+Date:   Mon Nov 15 23:50:57 2021 +0100
+
+    Upgrade OpenSSL to 1.1.1l to allow arm64 build
+
+diff --git a/cmake/deps.cmake b/cmake/deps.cmake
+index fc775527..2e7aee31 100644
+--- a/cmake/deps.cmake
++++ b/cmake/deps.cmake
+@@ -35,13 +35,13 @@ set(LLVM_ARCHIVE_SHA256
+     CACHE INTERNAL ""
+ )
+ 
+-# Commit for tag 1.1.1c
++# Commit for tag 1.1.1l
+ set(OPENSSL_URL
+-    "https://github.com/openssl/openssl/archive/97ace46e11dba4c4c2b7cb67140b6ec152cfaaf4.zip"
++    "https://github.com/openssl/openssl/archive/fb047ebc87b18bdc4cf9ddee9ee1f5ed93e56aff.zip"
+     CACHE INTERNAL "URL of OpenSSL archive to use."
+ )
+ set(OPENSSL_ARCHIVE_SHA256
+-    "f093df5cfd7521d8f6a09f250d7e69159d1001c47419130e806488de8a6312d8"
++    "04991d67b8be8f5f86ff01cc7f1cdf1501059beacecf5292a22f941102df0966"
+     CACHE INTERNAL ""
+ )
+ 
+diff --git a/deps/openssl/CMakeLists.txt b/deps/openssl/CMakeLists.txt
+index d2d4144d..f6aa58e1 100644
+--- a/deps/openssl/CMakeLists.txt
++++ b/deps/openssl/CMakeLists.txt
+@@ -99,6 +99,8 @@ elseif(OS_DARWIN)
+ 		set(OPENSSL_CONFIGURE_ARCH "darwin-i386-cc")
+ 	elseif(ARCH_X64)
+ 		set(OPENSSL_CONFIGURE_ARCH "darwin64-x86_64-cc")
++	elseif(ARCH_ARM64)
++		set(OPENSSL_CONFIGURE_ARCH "darwin64-arm64-cc")
+ 	endif()
+ endif()
+ 
+@@ -110,7 +112,7 @@ endif()
+ set(OPENSSL_PREFIX         "${CMAKE_CURRENT_BINARY_DIR}/openssl")
+ set(OPENSSL_SRC_DIR        "${OPENSSL_PREFIX}/src/openssl")
+ set(OPENSSL_INSTALL_DIR    "${OPENSSL_PREFIX}/src/openssl-install")
+-set(OPENSSL_VERSION        "1.1.1c")
++set(OPENSSL_VERSION        "1.1.1l")
```

Comment:
Kirill, does this have to exactly match our OpenSSL 1.1 port?

Reply:
@mascguy not at all. To be `1.1.x` is enough. If I haven't use OpenSSL PG, it may have OpenSSL 3.0 by dependencies inside `/opt/local/include/openssl` and it brokes build because it is linked agains embeded OpenSSL but uses headers from system one.

From another hand, embeded OpenSSL too old to support Apple's arm64 :)

=> simple upgrade hardcoded version seems a nice and easy hack. Much better than cross build via universal varian.

Thus, I've raised in upstream an idea to release a new version which included my patch to support OpenSSL 3.0; and I hope soon it happened.